version of ALLEGRO with ECAL barrel with cell corners projective in phi

giovannimarchiori opened the pull request:
BEGINRELEASENOTES
- New version of ALLEGRO detector o1_v03 with ecal barrel with 11 layers with cell corners projective along phi. No changes to the other sub detectors.
ENDRELEASENOTES

giovannimarchiori commented:
Hi @BrieucF thanks for having looked at the PR and spotted the little typo in the documentation. Is there anything else needed before merging?

Thanks, Giovanni

BrieucF commented:
Hi Giovanni, yes sorry, I started to review then had to do something else. Can you add a test for this new ALLEGRO version? https://github.com/key4hep/k4geo/blob/main/lcgeoTests/CMakeLists.txt#L90-L94

giovannimarchiori commented:
> Hi Giovanni, yes sorry, I started to review then had to do something else. Can you add a test for this new ALLEGRO version? https://github.com/key4hep/k4geo/blob/main/lcgeoTests/CMakeLists.txt#L90-L94

done

giovannimarchiori commented:
downstream-build errors are unrelated

The following tests FAILED:
	  2 - t_test_CLIC_o3_v14_reco_conformal (SEGFAULT)
	  3 - t_test_CLIC_o3_v14_vtx_makentp (Failed)
	  5 - t_test_FCCee_o1_v04_reco_conformal (Subprocess aborted)

giovannimarchiori commented:
I just added some comments to the C++ code that builds the LAr detector to make it a bit easier to follow the steps of the G4 construction, and some comments to the corresponding xml file to clarify the meaning of the vector with the layer thicknesses

BrieucF commented:
Can you also use the v03 constructor for upstream and calibration xml's?

giovannimarchiori commented:
Yes, when we have agreed on the rounding ;)

BrieucF commented:
Ha sorry, missed the last mail, taking a look

giovannimarchiori commented:
@BrieucF : I updated also the other xml files for v03. In the xml of v02 I clarified that the dimensions are electrode lengths scaled by calo dR/L(plane). In the cpp of v03 I also added a check that the electrode plane length in the xml is consistent within 0.5 mm with that calculated from calo R1, R2 and alpha

Let me check with Nicolas if he's OK with these non rounded numbers and then I think we're good to merge

BrieucF commented:
Thanks Giovanni! LGTM!

andresailer commented on detector/calorimeter/ECalBarrel_NobleLiquid_InclinedTrapezoids_o1_v03_geo.cpp:
```diff
@@ -0,0 +1,679 @@
+    }
+    layersTotalHeight += layer.repeat() * layer.thickness();
+  }
+  lLog << MSG::DEBUG << "Total electrode length from calorimeter xml description (cm): " << layersTotalHeight << endmsg;
```
```suggestion
  lLog << MSG::DEBUG << "Total electrode length from calorimeter xml description (cm): " << layersTotalHeight/dd4hep::cm << endmsg;
```

giovannimarchiori replied:
Fixed in latest commit

andresailer commented on detector/calorimeter/ECalBarrel_NobleLiquid_InclinedTrapezoids_o1_v03_geo.cpp:
```diff
@@ -0,0 +1,679 @@
+
+  // check that electrode length is consistent with calorimeter radial extent
+  // and inclination angle to within 0.5 mm
+  if (fabs(planeLength - layersTotalHeight) > 0.05) {
```
```suggestion
  if (fabs(planeLength - layersTotalHeight) > 0.05*dd4hep::cm) {
```

andresailer reviewed (commented):
One should also use units in the CPP implementation

giovannimarchiori commented:
Fixed in latest commit

giovannimarchiori commented:
Hello @andresailer, 
please let me know if I need to do further modifications or if the code can be merged as it is now.
Cheers,
  Giovanni

andresailer reviewed (approved):
Nothing else from my side.
Thanks!

BrieucF commented:
Thanks Giovanni and Andre!
